Source/driver/Castro.cpp

```diff
@@ -3485,6 +3489,67 @@ Castro::add_magnetic_e( MultiFab& Bx,
   }
 
 
+}
+
+void
+Castro::check_div_B( MultiFab& Bx,
+                     MultiFab& By, 
+                     MultiFab& Bz,
+                     MultiFab& State)
+{
+
+ 
+
+  ReduceOps<ReduceOpSum> reduce_op;
+  ReduceData<int> reduce_data(reduce_op);
+  using ReduceTuple = typename decltype(reduce_data)::Type;
+
+           
+#ifdef _OPENMP
+#pragma omp parallel
+#endif
+  for (MFIter mfi(State, TilingIfNotGPU()); mfi.isValid(); ++mfi)
+  {
+      const Box& box     = mfi.tilebox();
+      auto Bx_arr = Bx.array(mfi);
+      auto By_arr = By.array(mfi);
+      auto Bz_arr = Bz.array(mfi);
+
+      const auto dx = geom.CellSizeArray();
+
+      reduce_op.eval(box, reduce_data,
+      [=] AMREX_GPU_HOST_DEVICE (int i, int j, int k) noexcept -> ReduceTuple
+      {
+          
+          Real divB = (Bx_arr(i+1,j,k) - Bx_arr(i,j,k))/dx[0] +
+                      (By_arr(i,j+1,k) - By_arr(i,j,k))/dx[1] + 
+                      (Bz_arr(i,j,k+1) - Bz_arr(i,j,k))/dx[2];
+
+          Real magB = std::sqrt(Bx_arr(i,j,k) * Bx_arr(i,j,k) + 
+                                By_arr(i,j,k) * By_arr(i,j,k) +
+                                Bz_arr(i,j,k) * Bz_arr(i,j,k));
+                  
```

Comment:
I think we want magB to be cell-centered, so you should average the face centered B to centers and then compute mag B of that

Reply:
true, I just changed it